p2p/src/sync/mock_consensus.rs

```diff
@@ -0,0 +1,869 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): A. Altonen
+#![allow(clippy::new_without_default, clippy::result_unit_err)]
```

Comment:
I agree, especially with the first one.

Reply:
The `new_without_default` maybe. But `result_unit_err` should remain; it's important not to do that in real life code, contrary to mocks.

Reply:
Ok, I have mistaken `result_unit_err` for `unit_arg` which is the other one I find annoying.

The `unit_arg` lint suggests to replace this:
```rust
fn foo() -> Result<(), _> {
    ...
    Ok(function_that_returns_unit())
}
```
with:
```rust
fn foo() -> Result<(), _> {
    ...
    function_that_returns_unit();
    Ok(())
}
```

I do not find the latter to be an improvement. In case the return type of either `function_that_returns_unit` changes, the first version triggers a compilation error, which is sometimes useful.

Sorry for the mix-up and for being off topic.

Reply:
I don't find that `unit_arg` a big deal, and I personally prefer to see `Ok(())`. But if it's that annoying to you, sure, feel free to ask clippy to stop nagging about it globally.